Polarization

grburgess opened the pull request:
starting to add polarization capabilities.

giacomov commented on astromodels/core/polarization.py:
```diff
@@ -1,10 +1,102 @@
-    def __init__(self):
+    def __init__(self, type='linear'):
+
+        assert type in ['linear', 'stokes'], 'polarization must be linear or stokes'
```
Please do not use `type` as variable, as it is a python protected keyword.

grburgess replied:
Doh! Speed coding.

grburgess commented:
I fixed a bug and added imports. I'm not sure how you are doing the docs, so I just added an example in the Model tutorial docs that showed how to add polarization.

grburgess commented:
I'm curious if we should allow the creation of a polarization source directly in the point source constructor or keep it as a part of the spectral component. This latter is more explicit. If you could somehow have one polarization for each spectrum and measure them separately, that would be very cool. Thoughts @giacomov ?
